src/solutions/optimization_solutions.jl

```diff
@@ -34,11 +59,26 @@ function build_solution(prob::AbstractOptimizationProblem,
     T = eltype(eltype(u))
     N = ndims(u)
 
-    OptimizationSolution{T, N, typeof(u), typeof(prob), typeof(alg),
-                         typeof(minimum), typeof(original)}(u, prob, alg, minimum, retcode,
+    Base.depwarn("`build_solution(prob::AbstractOptimizationProblem, args...; kwargs...)` is deprecated." *
+                 " Consider implementing an `AbstractOptimizationCache` instead.",
+                 "build_solution(prob::AbstractOptimizationProblem, args...; kwargs...)")
```

Comment:
Because this is a warning, please make sure the upgrade path is ready to be merged by EOD and that we get the versions out. We shouldn't throw warnings from our own packages for more then 24 hours.